metrics/storage/blogbench.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Description of the test:
+# This test runs the 'blogbench', and extracts the 'scores' for reads
+# and writes
+# Note - the scores are *not* normalised for the number of iterations run,
+# they are total scores for all iterations (this is the blogbench default output)
+
+set -e
+
+# General env
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../lib/common.bash"
+TEST_ARGS="$@"
+
+TEST_NAME="blogbench"
+IMAGE="local-blogbench"
+DOCKERFILE="${SCRIPT_PATH}/Dockerfile"
+
+# Number of iterations for blogbench to run - note, results are not
+# scaled to iterations - more iterations results in bigger results
+ITERATIONS="${ITERATIONS:-30}"
+
+# Directory to run the test on
+TESTDIR="${TESTDIR:-/tmp}"
+CMD="blogbench -i ${ITERATIONS} -d ${TESTDIR}"
+
+# Check if we have our local image installed, and if not, install it
+# from the dockerfile
+function local_check_images() {
+	local exists=$(docker image inspect $IMAGE > /dev/null; echo $?)
+	[ $exists -eq 0 ] || docker build --label "$IMAGE" --tag "${IMAGE}:latest" "$DOCKERFILE"
+}
+
+function main() {
+	cmds=("awk")
+
+	init_env
+	check_cmds "${cmds[@]}"
+	local_check_images "$IMAGE"
+
+	metrics_json_init
+
+	local output=$(docker run --rm --runtime=$RUNTIME $IMAGE $CMD)
+
+	metrics_json_start_array
+
+	local writes=$(tail -2 <<< "$output" | head -1 | awk '{print $5}')
```

Comment:
`blogbench` outputs a bunch more info we are throwing away here. The code here was used for the `csv` file output, as that format was somewhat limited in what it could store - but our new shiny JSON format is much much more flexible.
Can you see if you can parse out some more info from the blogbench and store it in the JSON for us? Here (for others to see) is an example of a blogbench run:

```
root@80fbf6a05ff8:/# blogbench -i 5 -d /tmp

Frequency = 10 secs
Scratch dir = [/tmp]
Spawning 3 writers...
Spawning 1 rewriters...
Spawning 5 commenters...
Spawning 100 readers...
Benchmarking for 5 iterations.
The test will run during 0 minutes.

  Nb blogs   R articles    W articles    R pictures    W pictures    R comments    W comments
        57        61008          2995         44577          3313         13619          3584
        77        58250          1138         43798          1045         27073          3948
        95        55041          1279         39257           911         30923          3791
       109        52750          1188         37391           948         30166          3769
       126        52620          1152         37169           945         30842          3668

Final score for writes:           126
Final score for reads :          6021
```

So, I'd quite like to store those per-second statistics, as well as the config info.
I know, it would be nice if `blogbench` had a json output option, or output some other standard parseable format - but sadly I don't think it does.... :-(

Reply:
@grahamwhaley this is done too